File: common_core/src/main/java/com/xueh/comm_core/base/BaseActivity.java
package com.xueh.comm_core.base;

import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import androidx.appcompat.app.AppCompatActivity;


/**
 * @author: xuehao create time: 2017/7/26 11:29
 * tag: class//
 * description:三级统一结构型baseActivity
 */
public abstract class BaseActivity extends AppCompatActivity implements IBaseLogic {
    private static final String TAG = "BaseActivity";
    /**
     * activity Content view
     */
    private View inflateView;
    protected LayoutInflater inflater;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        getWindow().requestFeature(Window.FEATURE_CONTENT_TRANSITIONS);
        inflater = this.getLayoutInflater();
        inflateView = inflater.inflate(getCreateViewLayoutId(), null);
        setContentView(inflateView);
        super.onCreate(savedInstanceState);
        initDataBeforeView();
        initView(inflateView, savedInstanceState);
        initListener();
        initDataAfterView();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        inflateView = null;
        inflater = null;
    }
}

File: common_core/src/main/java/com/xueh/comm_core/base/BaseActivity.kt
package com.xueh.comm_core.base

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.view.Window
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding
import com.xueh.comm_core.helper.inflateBindingWithGeneric

/**
 * @author: xuehao create time: 2017/7/26 11:29
 * tag: class//
 * description:三级统一结构型baseActivity
 */
abstract class BaseActivity<VB : ViewBinding> : AppCompatActivity(), IBaseLogic {
    lateinit var binding: VB

    override fun onCreate(savedInstanceState: Bundle?) {
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        window.requestFeature(Window.FEATURE_CONTENT_TRANSITIONS)
        super.onCreate(savedInstanceState)
        binding = inflateBindingWithGeneric(layoutInflater)
        setContentView(binding.root)
        initDataBeforeView()
        initView(savedInstanceState)
        initListener()
        initDataAfterView()
    }
}